pkg/util/nichelper/helper.go

```diff
@@ -118,3 +118,180 @@ func currentNodeMatchesSelector(nodeName string, managedNodes string) (bool, err
 	}
 	return false, nil
 }
+
+func IsNICSriovCapable(deviceAddr string) (bool, error) {
+	_, err := os.Stat(filepath.Join(defaultDevicePath, deviceAddr, defaultTotalVFFile))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	return true, nil
+}
+
+func CurrentVFConfigured(deviceAddr string) (int, error) {
+	contents, err := os.ReadFile(filepath.Join(defaultDevicePath, deviceAddr, defaultConfiguredVFFile))
+	if err != nil {
+		return 0, fmt.Errorf("error reading configure vfs for device %s: %v", deviceAddr, err)
+	}
+
+	numvfs := strings.Trim(string(contents), "\n")
+	return strconv.Atoi(numvfs)
+}
+
+func ConfigureVF(deviceAddr string, numvfs int) error {
```

Comment:
This is a great first start for SR-IOV drivers that expose a sysfs interface, but there are others that we will probably need to support at some point, like ixgbe. The way to set the numvfs for that driver is by setting `max_vfs=N` when the driver is loaded.

There are two ways we can proceed: by making a shim driver that exposes a sysfs interface for this and similar drivers, or we can make a Go interface like:


```go
type SRIOVDriver interface {
    SetNumVFs(n int) error
}
```

And then make a two main types that implement that interface. We can make:

- `SRIOVSysFSDriver`
- `ModParamDriver` that unloads the driver, then reloads with with the `max_vfs` set

The downside of the second way is that my unloading the driver, we disrupt the other VFs. However, I haven't tried to check if the sysfs has the same property.